http-server/src/server.rs

```diff
@@ -301,11 +301,15 @@ impl<M: Middleware> Server<M> {
 					// Run some validation on the http request, then read the body and try to deserialize it into one of
 					// two cases: a single RPC request or a batch of RPC requests.
 					async move {
-						if let Err(e) = access_control_is_valid(&access_control, &request) {
-							return Ok::<_, HyperError>(e);
-						}
+						// Only `POST` and `OPTIONS` methods are allowed.
+						match *request.method() {
+							Method::POST if content_type_is_json(&request) => (),
+							Method::POST => return Ok::<_, HyperError>(response::unsupported_content_type()),
+							Method::OPTIONS => return Ok::<_, HyperError>(response::ok_response("".into())),
```

Comment:
I think we should also run `access_control_is_valid` in this OPTIONS branch because at least one of the checks it does it on a header only provided as part of the preflight request.